src/squlearn/util/optree/optree_evaluate.py

```diff
@@ -941,19 +942,11 @@ def change_order(n, reorder_list):
             if i[1] == n:
                 return i[0]
 
-    def maximum_decompose(circ):
-        """Helper function to decompose circuits."""
-        circ_dec = circ.decompose()
-        while circ_dec != circ:
-            circ_dec = circ_dec.decompose()
-            circ = circ.decompose()
-        return circ_dec
-
     if circ_in.num_clbits == 0:
         return circ_in.measure_all(inplace=False)
     else:
         qubits = [i for i in range(circ_in.num_qubits)]
-        circ_in = maximum_decompose(circ_in)
+        circ_in = DecomposeToStd(circ_in) # POTENTIAL UNNECESSARY
```

Comment:
Also here, what's up with the comment? If it is unnecessary, remove it, else remove the comment.

Reply:
Its a comment for David, whether all gates in the circuit at this point are already decomposed and therefore the decompose-function is not needed anymore at this point. I am not totally sure through which ways this function at this point is called and whether it always already passed a decomposer.

Reply:
Still, please don't use `TODO`-like comments in the code. Use the PR conversation for stuff like this 🙂